cli/console.py

```diff
@@ -1,2 +1,122 @@
+    container_name = azure_storage["azure_container_name"]
+    with console.status(
+        f"Fetching jobs in {container_name} container...\n",
+        spinner="aesthetic",
+    ):
+        sp_credential = obtain_sp_credential()
+        try:
+            blob_client = instantiate_blob_client(
+                sp_credential=sp_credential,
+                account_url=azure_storage["azure_storage_account_url"],
+            )
+            container_client = blob_client.get_container_client(container_name)
+            blob_list = container_client.list_blobs()
+            unique_jobs = get_unique_jobs_from_blobs(blob_list=blob_list)
+            console.print(unique_jobs)
+        except (ValueError, ResourceNotFoundError, ClientAuthenticationError) as e:
+            console.print(
+                "[italic red] :triangular_flag: Error instantiating blob client or finding specified resource."
+            )
+            raise e
+
+
+@app.command("list-tasks")
+def list_tasks(
+    job_id: Annotated[
+        str, typer.Option("--job-id", "-j", help="Job ID to list tasks for")
+    ],
+):
+    container_name = azure_storage["azure_container_name"]
+    with console.status(
+        f"Fetching tasks in {container_name} container for job_id {job_id}...\n",
+        spinner="aesthetic",
+    ):
+        sp_credential = obtain_sp_credential()
+        try:
+            blob_client = instantiate_blob_client(
+                sp_credential=sp_credential,
+                account_url=azure_storage["azure_storage_account_url"],
+            )
+            container_client = blob_client.get_container_client(container_name)
+            blob_list = container_client.list_blobs()
+            tasks_for_job = get_tasks_for_job_id(blob_list=blob_list, job_id=job_id)
+            console.print(tasks_for_job)
+        except (ResourceNotFoundError, ValueError, ClientAuthenticationError) as e:
+            console.print(
+                "[italic red] :triangular_flag: Error instantiating blob client or finding specified resource."
+            )
+            raise e
+
+
+@app.command("inspect-task")
+def inspect_task(
+    job_id: Annotated[
+        str, typer.Option("--job-id", "-j", help="Job ID to list tasks for")
+    ],
+    task_filename: Annotated[
+        str, typer.Option("--task-filename", "-t", help="Task filename to inspect")
+    ],
+):
+    with console.status(
+        f"Inspecting task in {task_filename} container for job_id {job_id}...\n",
+        spinner="aesthetic",
+    ):
+        sp_credential = obtain_sp_credential()
+        full_blob_path = f"{job_id}/{task_filename}"
+        try:
+            blob_client = instantiate_blob_client(
+                sp_credential=sp_credential,
+                account_url=azure_storage["azure_storage_account_url"],
+            )
+            blob_client = blob_client.get_blob_client(
+                container=azure_storage["azure_container_name"], blob=full_blob_path
+            )
```

Comment:
I'm guessing these two `blob_client`s are actually different types? If so, could we add type hints, or use different names?

Reply:
Yeah, it's confusing. The first one is a BlobServiceClient (allowing the SP to interact with all of Blob Storage) and the second one is the client for the blob itself. Will update 

https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-download-python